README.md

```diff
@@ -14,63 +14,155 @@ If you wish to deploy a local explorer, run:
+  saleFee: number // A fixed sale fee paid to Mesa
 }
 ```
 
-`templateId` is used to determine GraphQL schema -- either `EasyAuction` or `FixedPriceAuction` -- to create. `auction` address is the address of the newly deploed auction contract.
+### `SaleLauncher`
 
-### `AuctionLauncher`
+<sub>[`src/mappings/saleLauncher.ts`](src/mappings/saleLauncher.ts)</sub>
 
-<sub>[`src/mappings/auctionLauncher.ts`](src/mappings/auctionLauncher.ts)</sub>
+#### 1. `SaleInitialized`
+
+```typescript
+interface SaleInitialized {
+  sale: Address // Address of new Sale contract
+  templateId: number // The template used to create the sale
+  data: Bytes // Details of the sale as Bytes.
+}
+```
 
-WIP
+`templateId` is used to determine GraphQL schema -- either `FairSale` or `FixedPriceSale` -- to create. `sale` address is the address of the newly deploed sale contract.
 
 ### `TemplateLauncher`
 
 <sub>[`src/mappings/templateLauncher.ts`](src/mappings/templateLauncher.ts)</sub>
 
-WIP
+#### 1.`TemplateLaunched` from `TemplateLauncher.launchTemplate`
+
+```typescript
+interface TemplateLaunched {
+  sale: string // Address of the new *-Sale contract
+  templateId: string // Index of the template
+}
+```
+
+#### 2.`TemplateAdded` from `TemplateLauncher.addTemplate`
+
+```typescript
+interface TemplateAdded {
+  template: Address // Address of new Template contract
+  templateId: number // Index of the template
+}
+```
+
+#### 3.`TemplateRemoved` from `TemplateLauncher.removeTemplate`
 
-### `EasyAuction`
+Handles removing templates
 
-<sub>[`src/mappings/auctions/easyAuction.ts`](src/mappings/auctions/easyAuction.ts)
+```typescript
+interface TemplateRemoved {
+  template: Address // Address of new Template contract
+  templateId: number // Index of the template
+}
+```
+
+#### 4.`TemplateVerified` from `TemplateLauncher.verifyTemplate`
 
-### `FixedPriceAuction`
+Handles verifying templates - `TemplateVerified`
+
+```typescript
+interface TemplateVerified {
+  template: Address // Address of new Template contract
+  templateId: string // Index of the template
+}
+```
 
-<sub>[`src/mappings/auctions/fixedPriceAuction.ts`](src/mappings/auctions/fixedPriceAuction.ts)</sub>
+### `FairSale`
+
+<sub>[`src/mappings/sales/easySale.ts`](src/mappings/sales/fairSale.ts)
+
+#### 1. `SaleCleared` from `FairSale.clearSale`
+
+```typescript
+interface SaleCleared {
+  // No params
+}
+```
+
+#### 2. `NewOrder` from `FairSale.newOrder`
+
+```typescript
+interface NewOrder {
+  ownerId: number
+  orderTokenOut: number
+  orderTokenIn: number
+}
+```
+
+### `FixedPriceSale`
+
+<sub>[`src/mappings/sales/fixedPriceSale.ts`](src/mappings/sales/fixedPriceSale.ts)</sub>
+
+#### 1. `SaleClosed` from `FixedPriceSale.closeSale`
+
+```typescript
+interface SaleClosed {
+  // No params
+}
+```
+
+#### 2. `NewPurchase` from `FixedPriceSale.buyTokens`
+
+```typescript
+interface NewPurchase {
+  amount: number // Amount purchased
+  buyer: Address // EOA
```

Comment:
Why not "TokenBought", this would be more consistent with the naming convention you created?